### ISSUE TITLE
NielsenAppSDK-XC - 9.2.0.0 is not compliant with Apple Privacy requirement

### ISSUE THREAD

**Issue by elye** [NielsenDigitalSDK/nielsenappsdk-ios-dynamic-spm-global#3]
Hi Nielsen, we're using "'NielsenAppSDK-XC', '9.2.0.0'" for our iOS App.

We got an update from Apple on Nielsen SDK is not compliance with Apple Privacy requirement.

> ITMS-91064: Invalid tracking information - A PrivacyInfo.xcprivacy file contains invalid tracking information at the following path: “Frameworks/NielsenAppApi.framework/PrivacyInfo.xcprivacy”. NSPrivacyTracking must be true if NSPrivacyTrackingDomains isn’t empty. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, keys and values in your app’s privacy manifest must be valid. For more details about privacy manifest files, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files.

Can you have this ready up soon. Thanks!

**Comment by paulmcgrathparamount** [comment]
I'm seeing the same thing. This is a concerning issue due to the May 1st deadline.

**Comment by mhaviv** [comment]
Were seeing the same issue as well. I saw in the manifest that `NSPrivacyTracking` is set to `true` but their are no `NSPrivacyTrackingDomains` specified. Based on the email and docs it seems domains need to be set when marking `NSPrivacyTracking` as true

**Comment by octover** [comment]
Yep, manually editing and setting `NSPrivacyTracking` to `false` seems to get it through which is what it should be if the `NSPrivacyTrackingDomains` is empty. Probably not 100% correct, but in a pinch seems like my apps aren't getting rejected when submitting them for review now.

**Comment by paulmcgrathparamount** [comment]
I'm not sure altering the privacy files manually are the way to go here, wouldn't that violate the licensing agreement?

**Comment by romainbiard** [comment]
if you keep `NSPrivacyTracking` to `true` and add a fake domain in `NSPrivacyTrackingDomains` (like www.nielsen.com) it also works.
@akashnielsen @mpetrovnielsen @digappsdkiOS @digappsdkiosteam could you please have a look, we are all facing this issue and need an ETA to be able to publish our apps on the appstore

**Comment by romainbiard** [comment]
🎁 https://gist.github.com/romainbiard/63004122fa9d87119380a7db973aea23

**Comment by AmerigoM** [comment]
Is there any update on this matter from the authors? I'm facing this issue and it's actually concerning. I don't know if inserting a fake domain and taking responsibility for the SDK is the way to go, as @romainbiard suggested.

**Comment by mhaviv** [comment]
Tagging some of the authors: @digappsdkiOS @digappsdkiosteam @akashnielsen @mpetrovnielsen
This will likely hold back some of our releases and is a bit of a concern. Any update on if this is being looked into would be great

**Comment by jaggsu01** [comment]
Hi All,
We have worked on this last year itself when Apple introduced this feature, please follow the below steps to latch to the tracking domain(TD) flavor, while we are discussing to make this part of our default distributions going forward. Please reach out to your Nielsen client service representative if you face any further issue. 

# CocoaPods (Pod file)

    // Change Global flavor
    target 'NielsenVideoPlayerSwiftPodsXC' do
      pod 'NielsenAppSDK-XC'
    end
    
     // To Global-TD flavor
    target 'NielsenVideoPlayerSwiftPodsXC' do
      pod 'NielsenAppSDK-TD-XC'
    End

# Carthage (Cartfile)

      // Change Global flavor
     binary "https://raw.githubusercontent.com/nielsendigitalsdk/nielsenappsdk-ios-specs- 
     dynamic/master/Carthage/NielsenAppSDK/NielsenAppSDK.json"


    // To Global-TD flavor
    binary "https://raw.githubusercontent.com/nielsendigitalsdk/nielsenappsdk-ios-specs- 
       dynamic/master/Carthage/NielsenAppSDK-TD/NielsenAppSDK.json"


# Swift Package Manager using Local Package

   // Change Global Flavor
     .package(url: "https://github.com/NielsenDigitalSDK/nielsenappsdk-ios-dynamic-spm-global",
                 from: "9.2.0")
       .target(
            name: "LocalPackage",
            dependencies: [.product(name: "NielsenAppApi",
                                    package: "nielsenappsdk-ios-dynamic-spm-global")]
         ),

  // To Global-TD Flavor
     .package(url: "https://github.com/NielsenDigitalSDK/nielsenappsdk-ios-dynamic-spm-global-td”,
                 from: "9.2.0")
       .target(
            name: "LocalPackage",
            dependencies: [.product(name: "NielsenAppApi",
                                    package: "nielsenappsdk-ios-dynamic-spm-global-td”)]
         ),

# Swift Package Manager using Remote Package

// Change Global Flavor
https://github.com/NielsenDigitalSDK/nielsenappsdk-ios-dynamic-spm-global


// To Global-TD Flavor
https://github.com/NielsenDigitalSDK/nielsenappsdk-ios-dynamic-spm-global-td

**Comment by jaggsu01** [comment]
@octover @romainbiard @elye @mhaviv @paulmcgrathparamount Hope the above comment helps you all. We have decided to ship our tracking domain flavor as part of default distributions, but it may take time until this Friday. If you folks are blocked, feel free to use the above TD links and get your apps live in the store.

**Comment by romainbiard** [comment]
Thanks for the reply, will try this (move to TD package) and give you a feedback

**Comment by elye** [comment]
Hi @jaggsu01 , thanks for looking into it.

I have a question. What's the main differences between the Global Flavor vs the Global TD Flavor?  e.g. does the TD has more tracking?

**Comment by vgane1119** [comment]
@jaggsu01 I see latest release 9.3.3 - https://github.com/NielsenDigitalSDK/nielsenappsdk-ios-dynamic-spm-global, does this include the fix for tracking domain?

**Comment by jaggsu01** [comment]
@vgane1119 @elye All, Yes we have released v9.3.0.0 on May 10th, providing the tracking domain in privacy manifest file of the SDK. TD flavor had the same but we kept it as an adhoc solution, as we did not know the exact timelines from Apple when the mandate will be enforced. Since Apple has started the enforcement, we made the same changes in our standard distributions, so that clients can get future updates smoothly. For more info. https://engineeringportal.nielsen.com/wiki/iOS_SDK_Release_Notes
